### PR TITLE
Enable Platform session cache

### PR DESCRIPTION
**Pull request by peraltafederico**
Send session outputs to the Platform API.

### CODE REVIEW

**Comment by sourishkrout** [comment]
I'll do the review on this one.

File: src/extension/extension.ts [hunk 1]
```diff
@@ -112,6 +112,8 @@ export class RunmeExtension {
       ? new GrpcSerializer(context, server, kernel)
       : new WasmSerializer(context, kernel)
     this.serializer = serializer
+    kernel.registerCellTerminalState
```
Comment (sourishkrout) [inline src/extension/extension.ts:115]: what's this?

File: .github/scripts/overwrites/stateful.sh [hunk 2]
```diff
@@ -11,7 +11,7 @@ npm pkg set contributes.configuration[0].properties[runme.app.enableShare].defau
 npm pkg set contributes.configuration[0].properties[runme.experiments.escalationButton].default=true --json
 npm pkg set contributes.configuration[0].properties[runme.server.lifecycleIdentity].default=1 --json
 npm pkg set contributes.configuration[0].properties[runme.app.notebookAutoSave].default="yes"
-npm pkg set contributes.configuration[0].properties[runme.app.sessionOutputs].default=false --json
+npm pkg set contributes.configuration[0].properties[runme.app.sessionOutputs].default=true --json
```
Comment (sourishkrout) [inline .github/scripts/overwrites/stateful.sh:14]: if we just want default, which is `true,` perhaps we just remove the whole line altogether?

File: src/extension/messages/platformRequest/saveCellExecution.ts [hunk 3]
```diff
@@ -25,10 +26,18 @@ export default async function saveCellExecution(
...
+  // Save the file to ensure the serialization completes before saving the cell execution.
+  // This guarantees we access the latest cache state of the serializer.
+  await editor.notebook.save()
+
   log.info('Saving cell execution')
 
   const escalationButton = kernel.hasExperimentEnabled('escalationButton', false)!
   const sessionId = kernel.getRunnerEnvironment()?.getSessionId()
+  const cacheId = GrpcSerializer.getDocumentCacheId(editor.notebook.metadata) as string
+  const plainSessionOutput = await kernel.getSerializer()?.getPlainCache(cacheId)
```
Comment (sourishkrout) [inline src/extension/messages/platformRequest/saveCellExecution.ts:38]: Why not make `getPlainCache` and `getMaskedCache` public methods on `Kernel`? I don't love exposing the `serializer` instance directly.

**Review by sourishkrout — CHANGES_REQUESTED** [review]
Some suggestions and questions. Otherwise 👍.

File: src/extension/kernel.ts [hunk 4]
```diff
@@ -184,6 +185,14 @@ export class Kernel implements Disposable {
     this.category = category
   }
 
+  setSerializer(serializer: GrpcSerializer) {
+    this.serializer = serializer
+  }
+
+  getSerializer() {
```
Comment (sourishkrout) [inline src/extension/kernel.ts:192]: Interface not narrow enough, please see comment inside where saving is handled

File: src/extension/extension.ts (hunk 1 above)
Comment (peraltafederico) [reply src/extension/extension.ts]: Fixed in 2c69b84a17dcf327bd3ee5e4e305200fb220cece

File: .github/scripts/overwrites/stateful.sh (hunk 2 above)
Comment (peraltafederico) [reply .github/scripts/overwrites/stateful.sh]: Fixed in 2c69b84a17dcf327bd3ee5e4e305200fb220cece

File: src/extension/messages/platformRequest/saveCellExecution.ts (hunk 3 above)
Comment (peraltafederico) [reply src/extension/messages/platformRequest/saveCellExecution.ts]: Fixed in 2c69b84a17dcf327bd3ee5e4e305200fb220cece

File: src/extension/kernel.ts (hunk 4 above)
Comment (peraltafederico) [reply src/extension/kernel.ts]: Fixed in 2c69b84a17dcf327bd3ee5e4e305200fb220cece

**Review by sourishkrout — APPROVED** [review]
👍 LGTM

**Pull request merged**